src/components/UniversalNavbarExpanded/NavLink.js

```diff
@@ -35,11 +38,10 @@ const handleSamePage = ({
   ) {
     return
   }
-  let pathOnly = window.location.href.replace(window.location.origin, '')
-  if (window.location.search) {
-    pathOnly = pathOnly.replace(window.location.search, '')
-  }
-  if (pathOnly === href) {
+  const slashGlobalRegExp = /\//g
+  const strippedPath = window.location.pathname.replace(slashGlobalRegExp, '')
+  const strippedHref = href.replace(slashGlobalRegExp, '')
+  if (strippedPath === strippedHref) {
```

Comment:
A couple thoughts:

1. I can almost follow along by just reading, but not quite

1. I think it might be nice to share this in a helper, and, a really well-named function name could help make it self documenting

1. This sort of code really lends itself to unit tests since you can supply the input and assert on the outputs

Reply:
Fair points. I'll convert to a helper and add a couple quick tests.